src/phoenix/server/api/routers/v1/datasets.py

```diff
@@ -163,6 +165,102 @@ async def list_datasets(request: Request) -> Response:
         return JSONResponse(content={"next_cursor": next_cursor, "data": data})
 
 
+async def delete_dataset_by_id(request: Request) -> Response:
+    """
+    summary: Delete dataset by ID
+    operationId: deleteDatasetById
+    tags:
+      - datasets
+    parameters:
+      - in: path
+        name: id
+        required: true
+        schema:
+          type: string
+    responses:
+      200:
+        description: Success
+      403:
+        description: Forbidden
+      404:
+        description: Dataset not found
+      422:
+        description: Dataset ID is invalid
+    """
+    if id_ := request.path_params.get("id"):
+        try:
+            dataset_id = from_global_id_with_expected_type(
+                GlobalID.from_id(id_),
+                Dataset.__name__,
+            )
+        except ValueError:
+            return Response(
+                content=f"Invalid Dataset ID: {id_}",
+                status_code=HTTP_422_UNPROCESSABLE_ENTITY,
+            )
+    else:
+        return Response(
+            content="Missing Dataset ID",
+            status_code=HTTP_422_UNPROCESSABLE_ENTITY,
+        )
+    project_names_stmt = (
+        select(models.Experiment.project_name)
+        .where(models.Experiment.dataset_id == dataset_id)
+        .where(models.Experiment.project_name.isnot(None))
+    )
+    eval_trace_ids_stmt = (
+        select(models.ExperimentRunAnnotation.trace_id)
+        .join(models.ExperimentRun)
+        .join_from(models.ExperimentRun, models.Experiment)
+        .where(models.Experiment.dataset_id == dataset_id)
+        .where(models.ExperimentRunAnnotation.trace_id.isnot(None))
+    )
+    stmt = (
+        delete(models.Dataset).where(models.Dataset.id == dataset_id).returning(models.Dataset.id)
+    )
+    async with request.app.state.db() as session:
+        project_names = await session.scalars(project_names_stmt)
+        eval_trace_ids = await session.scalars(eval_trace_ids_stmt)
+        if (await session.scalar(stmt)) is None:
+            return Response(content="Dataset does not exist", status_code=HTTP_404_NOT_FOUND)
+    if not (project_names or eval_trace_ids):
+        return Response()
+    tasks = BackgroundTasks()
+    if project_names:
+        tasks.add_task(_delete_projects, request.app.state.db, frozenset(project_names))
+    if eval_trace_ids:
+        tasks.add_task(_delete_traces, request.app.state.db, frozenset(eval_trace_ids))
+    return Response(background=tasks)
```

Comment:
Would return at least the deleted dataset id as part of the response.

Reply:
i changed it to 204